### PR TITLE
Adding SIT calculations to the rate engine [delivers #156824380]

### PR DESCRIPTION
**Pull request by akostibas**
## Description

PPM should be compensated for Storage in Transit (SIT.) Thus, if the service member expects to be holding their goods in storage we need to be able to calculate the reimbursement.

For PPM, SIT only applies at the destination, and excludes the 1st day costs that are normally associated. [link](https://defensedigitalservice.slack.com/archives/G8HJRJBMZ/p1524160950000548)

## Code Review Verification Steps

* [ ] All tests pass.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Request review from a member of a different team.
* [ ] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/156824380) for this change

### CODE REVIEW

File: pkg/rateengine/nonlinehaul.go [hunk 1]
```diff
@@ -46,7 +46,39 @@ func (re *RateEngine) fullUnpackCents(cwt unit.CWT, zip3 string, date time.Time)
...
+
+	sa, err := models.FetchTariff400ngServiceAreaForZip3(re.db, zip3, date)
+	if err != nil {
+		return 0, err
+	}
+
+	var sitTotal unit.Cents
+
+	if isPPM {
+		sitTotal = unit.Cents(cwt.Int() * sa.SIT185BRateCents.Int() * daysInSIT)
```
Comment (jim) [inline pkg/rateengine/nonlinehaul.go]: This could be:

```go
sitTotal = sa.SIT185BRateCents.Multiply(daysInSIT).Multiply(cwt.Int())
```

To avoid some of the casting.

File: pkg/rateengine/nonlinehaul.go [hunk 2]
```diff
@@ -46,7 +46,39 @@ func (re *RateEngine) fullUnpackCents(cwt unit.CWT, zip3 string, date time.Time)
...
+	} else {
+		sitTotal = unit.Cents(cwt.Int() * sa.SIT185ARateCents.Int())
+		daysInSIT--
+		if daysInSIT > 0 {
```
Comment (jim) [inline pkg/rateengine/nonlinehaul.go]: I think that decrementing `daysInSIT` before this comparison makes the code a little hard to follow. It's pretty rare to see that outside a looping context. I'd suggest introducing a new variable with a helpful name (perhaps `additionalDays = daysInSIT - 1`) or just using `daysInSIT > 1` in the conditional.

File: pkg/rateengine/nonlinehaul_test.go [hunk 3]
```diff
@@ -151,6 +151,62 @@ func (suite *RateEngineSuite) Test_CheckFullUnpack() {
...
+	suite.mustSave(&sa)
+
+	// Test PPM SIT charges
+	charge, err := engine.sitCharge(cwt, daysInSIT, zip3, testdatagen.DateInsidePeakRateCycle, true)
+	if err != nil {
+		t.Fatalf("error calculating SIT charge: %s", err)
+	}
+	expected := sit185BRate.Multiply(daysInSIT).Multiply(cwt.Int())
+	if charge != expected {
+		t.Errorf("wrong non-linehaul charge total: expected %d, got %d", expected, charge)
```
Comment (jim) [inline pkg/rateengine/nonlinehaul_test.go]: I think this error message should reference `sitCharge`, not `non-linehaul`. There are a few more instances of this below.

File: pkg/rateengine/nonlinehaul.go [hunk 4]
```diff
@@ -46,7 +46,39 @@ func (re *RateEngine) fullUnpackCents(cwt unit.CWT, zip3 string, date time.Time)
 	return unit.Cents(math.Round(float64(cwt.Int()*fullUnpackRate) / 1000.0)), nil
 }
 
-func (re *RateEngine) nonLinehaulChargeTotalCents(weight unit.Pound, originZip5 string, destinationZip5 string, date time.Time) (unit.Cents, error) {
+// sitCharge calculates the SIT charge based on various factors.
+// If `isPPM` (Personally Procured Move) is True we do not apply the first-day
+// storage fees, 185A, to the total.
+func (re *RateEngine) sitCharge(cwt unit.CWT, daysInSIT int, zip3 string, date time.Time, isPPM bool) (unit.Cents, error) {
+	if daysInSIT <= 0 {
+		re.logger.Info("requested sitCharge for zero or less days in SIT?")
```
Comment (kilbergr) [inline pkg/rateengine/nonlinehaul.go]: Why is this a question?

File: pkg/rateengine/nonlinehaul_test.go [hunk 5]
```diff
@@ -151,6 +151,62 @@ func (suite *RateEngineSuite) Test_CheckFullUnpack() {
...
+
+	// Test HHG SIT charges
+	charge, err = engine.sitCharge(cwt, daysInSIT, zip3, testdatagen.DateInsidePeakRateCycle, false)
+	if err != nil {
+		t.Fatalf("error calculating SIT charge: %s", err)
+	}
+	expected = unit.Cents(
+		sit185ARate.Multiply(cwt.Int()).Int() + sit185BRate.Multiply(daysInSIT-1).Multiply(cwt.Int()).Int())
+	if charge != expected {
+		t.Errorf("wrong non-linehaul charge total: expected %d, got %d", expected, charge)
```
Comment (kilbergr) [inline pkg/rateengine/nonlinehaul_test.go]: same as Jim's comment above. Maybe specify this is HHG SIT.

File: pkg/rateengine/nonlinehaul.go (hunk 4 above)
Comment (akostibas) [reply pkg/rateengine/nonlinehaul.go]: How meta!
